RUBY-3384: Add service_manager role

brujeo opened the pull request:
Add Service manager role
https://eaflood.atlassian.net/browse/RUBY-3384

brujeo committed:
[RUBY-3384] Add service_manager role

timstone commented on spec/models/ability_spec.rb:
```diff
@@ -64,6 +64,17 @@
+    it_behaves_like "can manage users"
+    it_behaves_like "can manage registrations"
+
+    it { expect(ability).not_to be_able_to(:read, Reports::DefraQuarterlyStatsService) }
```
I'd like to double check this, it doesn't seem right.

brujeo committed:
[RUBY-3384] Adding read_quarterly_stats permission for service_manager role

brujeo committed:
[RUBY-3384] Add new permissions for charge adjustments and payments

brujeo committed:
[RUBY-3384] Add authorization for adding and refunding payments

This commit adds authorization checks for adding and refunding payments in the payments controller and record refunds controller. It also updates the ability model to include the new permissions. The changes ensure that only authorized users can perform these actions.

brujeo committed:
[RUBY-3384] Refactor user role in record_refunds_spec.rb

brujeo committed:
[RUBY-3384] Refactor user role in record_refunds_spec.rb and add authorization for adding and refunding payments

brujeo committed:
[RUBY-3384] Refactor SonarCloud analysis in ci.yml

brujeo committed:
[RUBY-3384] Refactor SonarCloud analysis in ci.yml

brujeo committed:
[RUBY-3384] Refactor SonarCloud analysis in ci.yml

brujeo committed:
[RUBY-3384] Refactor SonarCloud analysis in ci.yml

brujeo committed:
[RUBY-3384] adjusting SonarCloud config

brujeo committed:
[RUBY-3384] Refactor SonarCloud analysis and update coverage output in ci.yml
